elements/lisk-bft/src/finality_manager.ts

```diff
@@ -339,7 +343,12 @@ export class FinalityManager extends EventEmitter {
 
 		const bftBlockHeaders = stateStore.chain.lastBlockHeaders;
 		const { ledger } = await this._getVotingLedger(stateStore);
-		const chainMaxHeightPrevoted = this._calculateMaxHeightPrevoted(ledger);
+		// lastBlockHeaders are sorted by height desc
+		const lastMaxHeightPrevoted =
+			bftBlockHeaders.length > 0 && bftBlockHeaders[0].asset.maxHeightPrevoted
+				? bftBlockHeaders[0].asset.maxHeightPrevoted
```

Comment:
`BlockHeaders[0]` is the block header at the current tip of the chain?

Reply:
yes! `bftBlockHeaders` are sorted by height desc, so the first element will be the last block